### PR TITLE
Adding email domain to identify students

### PR DESCRIPTION
**Pull request by ConnorCartwright**
Hey guys @tarebyte @johndbritton 

I've been working on issue [#448](https://github.com/education/classroom/issues/448) and have made some good progress.

I'm just trying to work on the assignment invitation side and was hoping to get your opinions - I think the way I've used the two view files (`add_email` and `email_domains`) is very messy and needs tidying up

I've added in a new file `add.rb` which should use Octokit's `add_email` to add an email to the current users account, however it's throwing a 404 error at the moment so that's something I need to look into.

Sorry for the delay in getting this up, I've been reading through the Rails Tutorial book during commuting to try and get up to speed.

### CODE REVIEW

**Comment by tarebyte** [comment]
Hi @ConnorCartwright :wave: I should let me take a look tomorrow and I should be able to get you some feedback.

**Comment by ConnorCartwright** [comment]
Hey Mark @tarebyte :wave:

I've made a few commits to fix the build issues with Travis, apologies for the style problems.

I've been trying to get the Octokit `add_email` method to work still, I'm not sure if I've set up the client wrong in `add.rb` :confused:

I've tried passing it both a string and an array of strings but it still returns a 404 error for me:

image

I'm able to post a new email address using curl so I'm a bit lost as to why we're getting a 404.

`curl --data '["test@test.com"]' -u username -X POST https://api.github.com/user/emails`

Thank you for taking a look, I'll try and make the changes :zap::fast_forward:

File: db/migrate/20150608134150_create_assignments.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ def change
     create_table :assignments do |t|
       t.boolean    :public_repo,  default: true
       t.string     :title,        null: false
+      t.string  :emaildomain,      null: true
```
Comment (tarebyte) [inline db/migrate/20150608134150_create_assignments.rb]: How did `emaildomain` become `email_domain`? In the schema file

**Comment by tarebyte** [comment]
@ConnorCartwright so there could be a couple of reasons.

Here is a change I'd like to see, let's drop the `GitHub::Add` and add the method to `GitHub::User`

```diff
diff --git a/app/models/github_user.rb b/app/models/github_user.rb
index fc38643..b0d1f65 100644
--- a/app/models/github_user.rb
+++ b/app/models/github_user.rb
@@ -5,6 +5,12 @@ class GitHubUser
     @id     = id
   end
 
+  def add_email(email_address)
+    GitHub::Errors.with_error_handling do
+      @client.add_email(email_address)
+    end
+  end
+
   def authorized_access_token?
     GitHub::Errors.with_error_handling do
       Classroom.github_client.check_application_authorization(@client.access_token,
diff --git a/lib/github/add.rb b/lib/github/add.rb
deleted file mode 100644
index 3059949..0000000
--- a/lib/github/add.rb
+++ /dev/null
@@ -1,14 +0,0 @@
-# frozen_string_literal: true
-module GitHub
-  class Add
-    attr_accessor :add_client
-
-    def initialize(access_token)
-      @add_client = Octokit::Client.new(access_token: access_token)
-    end
-
-    def add_email(email_address)
-      @add_client.add_email(email_address)
-    end
-  end
-end
```

File: db/migrate/20150608134150_create_assignments.rb (hunk 1 above)
Comment (ConnorCartwright) [inline db/migrate/20150608134150_create_assignments.rb]: @tarebyte Not sure when this happened but I've updated the migrations now

**Comment by ConnorCartwright** [comment]
Hey Mark @tarebyte, 

I've moved the add_email method to the `GitHubUser` file. w

When trying to call add_email I get a GitHub::NotFound exception, `Resource could not be found on github.com`.

I'll take a look at this tonight - not sure if this still uses Octokit or needs a little change.

**Comment by tarebyte** [comment]
Sorry for not following up on this on. @ConnorCartwright I'm going to close this in favor of #614

**Pull request closed without merging**